src/modules/samples/cpp/tests/SampleTests.cpp

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <gtest/gtest.h>
+
+#include <Mmi.h>
+#include <Sample.h>
+
+namespace OSConfig::Platform::Tests
+{
+    TEST(SampleTests, SampleTest)
+    {
+        std::string value = "Sample C++ Module";
+
+        Sample* sample = new Sample(0);
+        ASSERT_NE(sample, nullptr);
+        ASSERT_EQ(MMI_OK, sample->SetValue(value));
+        ASSERT_STREQ(value.c_str(), sample->GetValue().c_str());
```

Comment:
The tests should execrcise the base class GetInfo, Get and Set to cover all the coding that is needed for the module. These methods should all have same signatures as MmiGetInfo, MmiGet and MmiSet. Refer to the pattern in modules.md.